sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1058,86 @@ def _eval_is_nonnegative(self):
                 return True
             elif k.is_even is False:
                 return  False
+
+class multinomial(CombinatorialFunction):
+    r"""Implementation of the multinomial coefficient.
+        They can be expressed in numerous ways, including as a product of binomial coefficients or of factorials:
+    .. math:: \frac{(k_1+k_2+\cdots+k_m)!}{k_1!\, k_2! \cdots k_m!} = {k_1\choose k_1}{k_1+k_2\choose k_2}\cdots{k_1+k_2+\cdots+k_m\choose k_m}
+
+    The multinomial coefficients have a direct combinatorial interpretation,
+    as the number of ways of depositing n distinct objects into m distinct bins,
+    with k1 objects in the first bin, k2 objects in the second bin, and so on.
+
+    Examples
+    ========
+
+    >>> from sympy import Symbol, Rational, multinomial, expand_func, symbols
+    >>> n = Symbol('n', integer=True, positive=True)
+
+    >>> multinomial(15, 8)
+    490314
+
+    >>> multinomial(n, -1)
+    0
+
+    Comparing Coefficents generated with the multinomial function:
+
+    >>> a, b, c = symbols('a b c')
+    >>> ((a+b+c)**3).expand(func=True)
+    a**3 + 3*a**2*b + 3*a**2*c + 3*a*b**2 + 6*a*b*c + 3*a*c**2 + b**3 + 3*b**2*c + 3*b*c**2 + c**3
+
+    The Coefficient when of 2 1 0 corresponding to the second term is 3
+
+    >>> multinomial(2,1,0)
+    3
+
+    >>> multinomial(Rational(5, 4), 3)
+    gamma(21/4)/(6*gamma(9/4))
+
+    >>> multinomial(n, 3)
+    gamma(n + 4)/(6*gamma(n + 1))
+
+    >>> multinomial(n, 3).expand(func=True)
+    n**3/6 + n**2 + 11*n/6 + 1
+
+    >>> expand_func(multinomial(n, 3))
+    (n + 1)*(n + 2)*(n + 3)/6
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Multinomial_theorem#Multinomial_coefficients
+    .. [2] http://functions.wolfram.com/GammaBetaErf/Multinomial/02/0001/
+
+    """
+
+    @classmethod
+    def eval(cls, *k):
+        k = [sympify(x) for x in k]
+        numerator = Add(*k)
+        if all(x.is_Integer for x in k) and numerator > 0:
```

Comment:
Is this right if k = (1,-2, 3) for which numerator is 2?
```python
>>> k = (1,-2,3)
>>> numerator = sum(k)
>>> factorial(numerator)/Mul(*(factorial(x) for x in k))
0
```

Reply:
Yupp ... [check it out](https://www.wolframalpha.com/input/?i=Multinomial%281%2C-2%2C+3%29 )

Reply:
This is anecdotal. In the polynomial interpretation, sum(k) == n and all ki are nonnegative. What is the interpretation of a negative ki?

Reply:
As binomial coefficients like `binomial(3, 4)`, `binomial(3, -1)` are defined as zero and is comprehended as a coefficient in front of the term `a**4 * b**-1` of `(a + b)**3` expanded (which does not exist and hence becoming zero)

A multinomial coefficient of `multinomial(1,-2, 3)` can be comprehended as a coefficient of 
`a * b**-2 * c**3` of `(a + b + c)**2` expanded.

Reply:
no ...  not all k<sub>i</sub> are non negative ... only the sum of all k<sub>i</sub> is non negative
if you're looking for a physical interpretation of -ve ki then you might wanna take a look at [https://arxiv.org/pdf/1105.3689.pdf](https://arxiv.org/pdf/1105.3689.pdf)

Reply:
> This is anecdotal. In the polynomial interpretation, sum(k) == n and all ki are nonnegative. What is the interpretation of a negative ki?

it is an artifact of generalisation 😄 !

Reply:
OK, I understand.